### DISCUSSION TITLE
DeviceDiscovery does not appear to work in ifm3d v1.4.3

### DISCUSSION THREAD

**Discussion by aludman** [ifm/ifm3d#441] [General]
We have recently switched from ifm3d v0.93.0 to v1.4.3 and everything seems to be working except for `ifm3d::Device::DeviceDiscovery()` (or its command line equivalent). We've had no issues with device discovery while using the old sdk.

tcpdump shows that a request is sent (broadcast) on each of the NICs and a response is received from each sensor but the source ip of the response packets is 0.0.0.0, which the code in [discover.hpp](https://github.com/ifm/ifm3d/blob/v1.4.3/modules/device/src/libifm3d_device/discovery.hpp) is not set up to receive.

Comparing the [discover.hpp (v1.4.3)](https://github.com/ifm/ifm3d/blob/v1.4.3/modules/device/src/libifm3d_device/discovery.hpp) to [discover.hpp v0.93.0](https://github.com/ifm/ifm3d/blob/v0.93.0/modules/camera/src/libifm3d_camera/discovery.hpp) shows that the code in the older sdk (v0.93.0) used a second socket, which was not bound to any of the NICs, to listen for the response. For some reason the new sdk (v1.4.3) is missing this code.

![image](https://github.com/ifm/ifm3d/assets/69987355/4968ae98-2885-4074-a21d-8612855a01de)

I can get around the problem by using the v0.93.0 discover.hpp for discovery but I was wondering if others were able to use the DeviceDiscovery of the new sdk and if so, what am I doing differently.

We're using Ubuntu Server 22.04 LTS on a system with 2 physical NICs but got the same results on Ubuntu 18.04 Server.

Here is the tcpdump command I used and the output from it:
```
tcpdump -nnvXSs 0  -i enp1s0 udp port 3321
```
[tcpdump_u18_cli.log](https://github.com/ifm/ifm3d/files/13956292/tcpdump_u18_cli.log)

Has anyone else hit this issue?

**Comment by lola-masson** [comment]
Hi @aludman,

I am sorry it seems like your question fell through the cracks. So you still need help with the `DeviceDiscovery()`?
